sdk/src/utils.ts

```diff
@@ -6,9 +6,8 @@ export function logAndThrow(message: string): never {
 
 export function parseJSON(json: string): any {
     function revive(key: string, value: any, context: any) {
-        if (Number.isInteger(value)) {
+        if (Number.isInteger(value) && value.length > 15) {
```

Comment:
We should fix this for real.

But in the meantime, if we want to go with the "conditionally returns bigint if it's too big" logic, then it should be like this:

```js
if (Number.isInteger(value) && !Number.isSafeInteger(value)) {
```

Reply:
This was an oversight, thanks for catching it.